src/utils.js

```diff
@@ -163,6 +152,93 @@ function accumulateResults(url, options, property, soFar, totalLimit, token) {
     });
 }
 
+async function yieldResults(url, options) {
+    let {
+        signal,
+        responseKeyName,
+        totalRequestSize,
+        yieldRequestProcessor,
+        formatter
+    } = options;
+
+    if (aborted(signal)) {
+        return [];
+    }
+
+    const yieldRequestHistory = getHistoryYieldRequest(signal);
+    let count = 0;
+    let stop = false;
+    let newUrl = url;
+    let token;
+    let result;
+    let data;
+    let formattedData;
+
+    while (!stop) {
+        result = await yieldRequestHistory.next(newUrl).value;
+        data = result[responseKeyName];
+
+        if (data) {
+            count += data.length;
+            token = result.continuationToken;
+            formattedData = formatter(data);
+
+            if (token) {
+                yieldRequestProcessor(formattedData);
+                newUrl = addTokenToUrl(url, token);
+                yieldRequestHistory.next();
+            } else {
+                return formattedData;
+            }
+
+            if (aborted(signal) || count >= totalRequestSize) {
+                stop = true;
+            }
+        } else {
+            stop = true;
+        }
+    }
+
+    yieldRequestHistory.return(false);
+
+    return [];
+
+}
+
+function getHistoryYieldRequest(signal) {
+
+    function* yieldRequestHistory() {
+        let url = true;
+
+        while (url) {
+            url = yield;
+            yield fetch(encodeURI(url, { signal}))
+                .then(res => res.json());
+        }
+    }
+
+    const generator = yieldRequestHistory();
+    generator.next(); // prime
+
+    return generator;
+}
+
+function addTokenToUrl(url, token) {
+    if (token !== undefined) {
+        if (url.indexOf('?') < 0) {
```

Comment:
JavaScript has builtin API for this.

eg.
```
const urlObject = new URL(url);
urlObject.searchParams.set("next", token);

return urlObject.toString();
```

Reply:
Nice!